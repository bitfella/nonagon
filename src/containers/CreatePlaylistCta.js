import React, { Component } from 'react';
import { connect } from 'react-redux';
import { debounce } from 'lodash-es';
import styled from 'styled-components/macro';
import PropTypes from 'prop-types';
import createPlaylist from 'services/createPlaylist';

const StyledCreatePlaylistCta = styled.button`
  flex-grow: 1;
  padding: 1rem;
  color: ${props => props.theme.colors.a};
  font-size: 0.875rem;
  font-weight: 700;
  letter-spacing: 0.125rem;
  background-color: ${props => props.theme.colors.d};
  border: none;
  border-radius: 500px;
  cursor: pointer;

  &:disabled {
    cursor: progress;
    opacity: 0.5;
  }
`;

export class CreatePlaylistCta extends Component {
  static propTypes = {
    createPlaylist: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.state = {
      isCtaClicked: false
    };
  }

  manageCtaClick = () => {
    this.setState({
      isCtaClicked: true
    });
    this.props.createPlaylist();
  };

  render() {
    return (
      <StyledCreatePlaylistCta
        disabled={this.state.isCtaClicked}
        onClick={debounce(this.manageCtaClick, 1000, {
          leading: true,
          trailing: false
        })}
      >
        <span>CREATE PLAYLIST</span>
      </StyledCreatePlaylistCta>
    );
  }
}

const mapDispatchToProps = {
  createPlaylist
};

export default connect(
  null,
  mapDispatchToProps
)(CreatePlaylistCta);
