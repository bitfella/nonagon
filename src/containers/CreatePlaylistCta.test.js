import React from 'react';
import { shallow } from 'enzyme';
import { CreatePlaylistCta } from './CreatePlaylistCta';

const props = {
  createPlaylist: jest.fn()
};

describe('CreatePlaylistCta', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CreatePlaylistCta {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
