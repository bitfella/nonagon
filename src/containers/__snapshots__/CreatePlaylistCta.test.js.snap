// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlaylistCta renders correctly 1`] = `
<CreatePlaylistCta__StyledCreatePlaylistCta
  disabled={false}
  onClick={[Function]}
>
  <span>
    CREATE PLAYLIST
  </span>
</CreatePlaylistCta__StyledCreatePlaylistCta>
`;
